### DISCUSSION TITLE
Why docker run command doesn't work

### DISCUSSION THREAD

**Discussion by hanhanwu** [bentoml/BentoML#1143] [Q&A] [answered]
I'm following every step in the quick start: https://colab.research.google.com/github/bentoml/BentoML/blob/master/guides/quick-start/bentoml-quick-start-guide.ipynb

Also new to docker. I'm using Windows 10. Want to see how to run the built image through docker.
Every step till `bentoml containerize IrisClassifier:latest -t iris-classifier` works fine in my local ipython notebook.
A folder had been built in my local "\bentoml\repository\IrisClassifier\20200929232341_4535FA\".

But I tried to run command `docker run -p 5000:5000 iris-classifier:latest --workers=1 --enable-microbatch` through ipython or powershell, often got this error:
![error](https://user-images.githubusercontent.com/4024769/94654102-6f050900-02b1-11eb-905d-ec3aa97d4e9f.PNG)

Do you know how can I run this docker command successfully?

**Comment by yubozhao** [comment, marked as answer]
@hanhanwu Thank you for point it out.  We missed the `:latest` on the `bentoml containerize` command. It should be:
```bentoml containerize IrisClassifier:latest -t iris-classifier:latest```

In your case, the docker image is properly built.  You can use `docker images` command to list all of the docker images on your system, and you should see it in there. The docker image tag will be your IrisClassifier version `20200928232341_4535FA`. 

If you see the image, you can run ```docker run -p 5000:5000 iris-classifier:20200928232341_4535FA --workers=1 --enable-microbatch``` to test it out.

Right now, the `bentoml containerize` behavior is if you provide a docker image tag without `:`, it will default to use the bento's version instead. For this case, with `--tag iris-classifier`, the finished docker tag will be `iris-classifier:BENTO_VERSION`

**Reply by hanhanwu** [reply]
Great! Thank you!
Now I'm able to see the image running in the docker! 

![Capture](https://user-images.githubusercontent.com/4024769/94767376-0a08ec00-0361-11eb-8d22-255d2daea7c4.PNG)

**One more question, then how can I use this running image? Can I use it the same as REST API to give it some input and get the output? Or how can I run the trained model through docker?**

**Reply by yubozhao** [reply]
@hanhanwu Awesome, glad you have it running

Yes, you can use the running docker container as REST API server. If you have curl installed on your system. You can run the same curl command.

curl -i \
--header "Content-Type: application/json" \
--request POST \
--data '[[5.1, 3.5, 1.4, 0.2]]' \
localhost:5000/predict
You can also visit the index page http://localhost:5000 use the swagger UI.

how can I use this running image

     Once you build a docker image, you can deploy to a lot of places such as Kubernetes, AWS ECS and etc.  
     For your running image (container), it only work locally to your system, unless you have open your system for 
     others to access over network.
how can i run the trained model through docker

     I am not so sure I understand your question, can you elaborate?

**Reply by hanhanwu** [reply]
Perfect, this did answered my question. Thank you very much @yubozhao !
